source/text/glyphvertices.ts

```diff
@@ -1,128 +1,145 @@
+        let offset = 0;
+
+        /* tslint:disable-next-line:prefer-for-of */
+        for (let index = 0; index < array.length; ++index) {
+            const vertices = array[index];
+            if (vertices === undefined || vertices!.length === 0) {
+                continue;
+            }
+            concatenated._texCoords.set(vertices._texCoords, offset * GlyphVertices.FLOATS_PER_TEXCOORD);
+            concatenated._origins.set(vertices._origins, offset * GlyphVertices.FLOATS_PER_ORIGIN);
+            concatenated._tangents.set(vertices._tangents, offset * GlyphVertices.FLOATS_PER_TANGENT);
+            concatenated._bitangents.set(vertices._bitangents, offset * GlyphVertices.FLOATS_PER_BITANGENT);
+
+            offset += vertices.length;
         }
+        return concatenated;
     }
 
+
     /**
-     * Updates its buffers origins, tangents, ups and texCoords. Call this to get buffers suitable for LabelGeometry.
+     * Constructs a specialized arrays representing glyph vertex data.
+     * @param numberOfGlyphs - The number of glyph vertices required (number of glyphs).
      */
-    update(): void {
-        const length = this.length;
+    constructor(numberOfGlyphs: number) {
+        const length = Math.max(0, numberOfGlyphs);
 
-        this._origins = new Float32Array(length * 3);
-        this._tangents = new Float32Array(length * 3);
-        this._ups = new Float32Array(length * 3);
-
-        this._texCoords = new Float32Array(length * 4);
-
-        for (let i = 0; i < length; i++) {
-            const vertex = this._vertices[i];
-
-            this._origins[i * 3 + 0] = vertex.origin[0];
-            this._origins[i * 3 + 1] = vertex.origin[1];
-            this._origins[i * 3 + 2] = vertex.origin[2];
+        this._texCoords = new Float32Array(length * GlyphVertices.FLOATS_PER_TEXCOORD);
+        this._origins = new Float32Array(length * GlyphVertices.FLOATS_PER_ORIGIN);
+        this._tangents = new Float32Array(length * GlyphVertices.FLOATS_PER_TANGENT);
+        this._bitangents = new Float32Array(length * GlyphVertices.FLOATS_PER_BITANGENT);
+    }
 
-            this._tangents[i * 3 + 0] = vertex.tangent[0];
-            this._tangents[i * 3 + 1] = vertex.tangent[1];
-            this._tangents[i * 3 + 2] = vertex.tangent[2];
+    /**
+     * Efficiently reduces the size of all underlying float arrays without copying any data.
+     * @param numberOfGlyphs - Target number of glyphs to reduce the vertices to.
+     */
+    shrink(numberOfGlyphs: number): void {
+        const length = Math.max(0, numberOfGlyphs);
+
+        /* Utility that slices the underlying buffer to the required length in bytes, and then creates a new
+        float 32 array view to that buffer. */
+        const shrink = (array: Float32Array, length: number, components: number) =>
+            array = new Float32Array(
+                array.buffer.slice(0, length * array.BYTES_PER_ELEMENT * components),
```

Comment:
This doesn't seem to match the doc comment - `slice` always copies data (-> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer/slice).
For avoiding copies, use the 3-argument constructor (`new Float32Array(buffer [, byteOffset [, length]]);`).